bundle/edu.gemini.ags/src/main/scala/edu/gemini/ags/gems/GemsVoTableCatalog.scala

```diff
@@ -136,11 +136,14 @@ object GemsVoTableCatalog {
 
   // Sets the min/max magnitude limits in the given query arguments
   protected [gems] def optimizeMagnitudeLimits(criterions: List[GemsCatalogSearchCriterion]): List[MagnitudeConstraints] = {
+    val magConstraints = for {
+        criteria <- criterions
+      } yield criteria.criterion.magConstraints
```

Comment:
I'm partial to explicitly using `map` and `flatMap` unless you start to chain together too many in one expression.  So I would write that:

`val magConstraints = criterions.map(_.criterion.magConstraints)`

but it is debatable which is clearer.  In the steps that follow though, `magConstraints` get flattened twice.  Could we do that once here?  Now I might break out `for`:

````scala
val magConstraints = for {
  criteria <- criterions
  mc       <- criteria.criterion.magConstraints
} yield mc
````